app/models/post.rb

```diff
@@ -1,2 +1,14 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+  after_save :increment_posts_counter
+
+  def increment_posts_counter
+    user.increment!(:posts_counter)
+  end
```

Comment:
- [ ] Kindly change the `user` to `author` as the name is the `belongs_to` relation is `author`.